internal/dag/policy.go

```diff
@@ -311,3 +311,52 @@ func prefixReplacementsAreValid(replacements []contour_api_v1.ReplacePrefix) (st
 
 	return "", nil
 }
+
+func rateLimitPolicy(in *contour_api_v1.RateLimitPolicy) (*RateLimitPolicy, error) {
+	if in == nil || in.Local == nil {
+		return nil, nil
+	}
+
+	if in.Local.Requests <= 0 {
+		return nil, fmt.Errorf("invalid requests value %d in local rate limit policy", in.Local.Requests)
+	}
+
+	var fillInterval time.Duration
+	switch in.Local.Unit {
+	case "second":
+		fillInterval = time.Second
+	case "minute":
+		fillInterval = time.Minute
+	case "hour":
+		fillInterval = time.Hour
+	default:
+		return nil, fmt.Errorf("invalid unit %q in local rate limit policy", in.Local.Unit)
+	}
+
+	rp := &RateLimitPolicy{
```

Comment:
Might need one more additional validation that the configured fill_interval is more than `50ms`. 

ref: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/local_ratelimit/v3/local_rate_limit.proto#envoy-v3-api-field-extensions-filters-network-local-ratelimit-v3-localratelimit-token-bucket

Reply:
We should be OK since users can't configure `fill_interval` directly, it's inferred based on the `unit`, which has a smallest allowed value of `second`.

Reply:
Nice!